internal/util/util.go

```diff
@@ -433,3 +445,18 @@ func recreateVolumeSnapshotContent(vsc snapshotv1api.VolumeSnapshotContent, back
 
 	return nil
 }
+
+func DeleteVolumeSnapshotIfAny(ctx context.Context, snapshotClient snapshotterClientSet.Interface,
+	vs snapshotv1api.VolumeSnapshot, log logrus.FieldLogger) {
+	_, err := snapshotClient.SnapshotV1().VolumeSnapshots(vs.Namespace).Get(ctx, vs.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.WithError(err).Errorf("fail to get VolumeSnapshot %s/%s", vs.Namespace, vs.Name)
```

Comment:
This is not an error log necessarily since this function is called by "IfAny", so suggest to change its level, e.g., debug level.
Otherwise, since the VS must exist in the context this function is called, we just modify the function to "
DeleteVolumeSnapshot" and log error message on any errors? then we don't need to check `apierrors.IsNotFound(err)`

Reply:
This function is called to clear related VolumeSnapshot when DataUpload creation failed.
Whether it fails or not, the BIA is returning with the previous error.
I think setting the log level to info or warn should be enough in this case.
Users can see the information if VolumeSnapshot is not cleaned.